OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -24,133 +24,168 @@
 #include "MarkersReference.h"
 #include <OpenSim/Common/Units.h>
 #include <SimTKcommon/internal/State.h>
+#include <cmath>
 
 using namespace std;
 using namespace SimTK;
 
 namespace OpenSim {
 
-MarkersReference::MarkersReference() : Reference_<SimTK::Vec3>()
-{
+MarkersReference::MarkersReference() :
+    Reference_<SimTK::Vec3>() {
     constructProperties();
     setAuthors("Ajay Seth");
 }
 
 MarkersReference::MarkersReference(const std::string& markerFile,
-    Units modelUnits) : MarkersReference()
-{
+                                   Units modelUnits) :
+    MarkersReference() {
     loadMarkersFile(markerFile, modelUnits);
 }
 
-
-MarkersReference::MarkersReference(MarkerData *markerData,
-    const Set<MarkerWeight>* markerWeightSet) : MarkersReference()
-{
-    _markerData.reset(markerData);
-    if (markerWeightSet!=nullptr) upd_marker_weights()= *markerWeightSet;
-    populateFromMarkerData(*markerData);
+MarkersReference::
+MarkersReference(const TimeSeriesTable_<SimTK::Vec3>& markerTable,
```

Comment:
You may be right that the problem already existed, but the `MarkerData` had `Units` as a field and it was responsible for providing markers in SI units when asked for its data. Now that we have replaced it with a `Table` (which also maintains the units via its metadata) we need to make sure the units are handled correctly. It is definitely more flexible to pass the `MarkersReference` a `Table` and it should sensibly convert to internal (SI) units as occurs when loading from file. A straightforward solution would to have the Table-based constructor do the conversion and have the file-based constructor read a Table from file and delegate it to the Table-based constructor. Does that sound reasonable?

We should not dismiss this issue and either address it in this PR or a subsequent one.